spring-context/src/main/java/org/springframework/context/support/PropertySourcesPlaceholderConfigurer.java

```diff
@@ -172,6 +172,10 @@ protected ConfigurablePropertyResolver createPropertyResolver(MutablePropertySou
 	protected void processProperties(ConfigurableListableBeanFactory beanFactoryToProcess,
 			ConfigurablePropertyResolver propertyResolver) throws BeansException {
 
+		if(this.environment instanceof ApplicationServletEnvironment) {
+			propertyResolver.setConversionService(((ApplicationServletEnvironment) this.environment).getConversionService());
+		}
+		
```

Comment:
```suggestion
		if (this.environment instanceof ConfigurableEnvironment configurableEnvironment) {
			propertyResolver.setConversionService(configurableEnvironment.getConversionService());
		}
```

I looked into a possible fix before looking at your proposal, and I first came up with the analogous solution above.

We cannot use `ApplicationServletEnvironment` here, since that class is specific Spring Boot and Servlet deployments.

Thus, downcasting to `ConfigurableEnvironment` using `instanceof` pattern matching would be the correct way to do that.

Reply:
Thanks something didn't feel right with the change I made